language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/createvar/CreateVariableWithTypeCodeAction.java

```diff
@@ -0,0 +1,289 @@
+     *
+     * @param typeSymbol Return type descriptor of the method.
+     * @param context    CodeActionContext
+     * @return {@link List<String>}
+     */
+    private List<String> getPossibleTypes(TypeSymbol typeSymbol, CodeActionContext context) {
+        typeSymbol = getRawType(typeSymbol, context);
+        Set<String> possibleTypes = new HashSet<>();
+        List<TypeSymbol> errorTypes = new ArrayList<>();
+        if (typeSymbol.typeKind() == TypeDescKind.UNION) {
+            ((UnionTypeSymbol) typeSymbol)
+                    .memberTypeDescriptors()
+                    .stream()
+                    .map(memberTypeSymbol -> getRawType(memberTypeSymbol, context))
+                    .forEach(memberTypeSymbol -> {
+                        if (memberTypeSymbol.typeKind() == TypeDescKind.UNION) {
+                            possibleTypes.addAll(
+                                    ((UnionTypeSymbol) memberTypeSymbol)
+                                            .memberTypeDescriptors()
+                                            .stream()
+                                            .map(memberTSymbol -> getRawType(memberTSymbol, context))
+                                            .map(symbol -> getTypeName(symbol, context))
+                                            .collect(Collectors.toList()));
+                        } else if (memberTypeSymbol.typeKind() == TypeDescKind.ERROR ||
+                                CommonUtil.getRawType(memberTypeSymbol).typeKind() == TypeDescKind.ERROR) {
+                            errorTypes.add(memberTypeSymbol);
+                        } else {
+                            possibleTypes.add(getTypeName(memberTypeSymbol, context));
+                        }
+                    });
+        } else {
+            String type = getTypeName(typeSymbol, context);
+            if (!"any".equals(type)) {
+                return Collections.singletonList(type);
+            }
+        }
+
+        if (!errorTypes.isEmpty()) {
+            String errorTypeStr = errorTypes.stream()
+                    .map(type -> getTypeName(type, context))
+                    .collect(Collectors.joining("|"));
+            return possibleTypes.stream()
+                    .filter(type -> !"any".equals(type))
+                    .map(type -> type + "|" + errorTypeStr)
+                    .collect(Collectors.toList());
+        }
+        return possibleTypes.stream()
+                .filter(type -> !"any".equals(type))
+                .collect(Collectors.toList());
+    }
+
+    private Boolean isInRemoteMethodCallOrResourceAccess(CodeActionContext context) {
+        Node evalNode = context.nodeAtRange();
+        int count = 0;
+        while (evalNode.kind() != SyntaxKind.CLIENT_RESOURCE_ACCESS_ACTION
+                && evalNode.kind() != SyntaxKind.REMOTE_METHOD_CALL_ACTION) {
+            count++;
```

Comment:
Do we really need this `count` here? Because, if there's a client resource access or remote method call as a parent, it means the cursor is inside one of them

Reply:
Yes, that's true. But there is no need of going all the way up to the root node.

Reply:
No problem then.